pkg/referrerstore/oras/authprovider/k8secret_authprovider.go

```diff
@@ -0,0 +1,222 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authprovider
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/docker/cli/cli/config"
+	core "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+type k8SecretProviderFactory struct{}
+type k8SecretAuthProvider struct {
+	ratifyNamespace  string
+	config           k8SecretAuthProviderConf
+	clusterClientSet *kubernetes.Clientset
+}
+
+type secretConfig struct {
+	SecretName string `json:"secretName"`
+	Namespace  string `json:"namespace,omitempty"`
+}
+
+type k8SecretAuthProviderConf struct {
+	Name               string         `json:"name"`
+	ServiceAccountName string         `json:"serviceAccountName,omitempty"`
+	Secrets            []secretConfig `json:"secrets,omitempty"`
+}
+
+const defaultName = "default"
+const ratifyNamespaceEnv = "RATIFY_NAMESPACE"
+
+// init calls Register for our k8s-secrets provider
+func init() {
+	Register("k8s-secrets", &k8SecretProviderFactory{})
+}
+
+// Create returns a k8AuthProvider instance after parsing auth config and resolving
+// named K8 secrets
+func (s *k8SecretProviderFactory) Create(authProviderConfig AuthProviderConfig) (AuthProvider, error) {
+
+	conf := k8SecretAuthProviderConf{}
+	authProviderConfigBytes, err := json.Marshal(authProviderConfig)
+	if err != nil {
+		return nil, err
```

Comment:
There is a few error blocks in this method, can we wrap the error with more  details on which step failed?

Reply:
Yes I will wrap the errors with a message indicating the step it failed in